SU2_DEF/src/drivers/CDiscAdjDeformationDriver.cpp

```diff
@@ -0,0 +1,945 @@
+    }
+
+    if (rank == MASTER_NODE) cout << "\n------------------------ Mesh sensitivity Output ------------------------" << endl;
+    SetSensitivity_Files(geometry_container, config_container, nZone);
+
+    /*--- Initialize structure to store the gradient ---*/
+    Gradient = new su2double*[config_container[ZONE_0]->GetnDV()];
+
+    for (auto iDV = 0u; iDV  < config_container[ZONE_0]->GetnDV(); iDV++) {
+        /*--- Initialize to zero ---*/
+        Gradient[iDV] = new su2double[config_container[ZONE_0]->GetnDV_Value(iDV)]();
+    }
+
+    Gradient_file.precision(driver_config->OptionIsSet("OUTPUT_PRECISION") ? driver_config->GetOutput_Precision() : 6);
+
+    /*--- For multizone computations the gradient contributions are summed up and written into one file. ---*/
+    for (iZone = 0; iZone < nZone; iZone++){
+        if ((config_container[iZone]->GetDesign_Variable(0) != NONE) &&
+            (config_container[iZone]->GetDesign_Variable(0) != SURFACE_FILE)) {
+
+        if (rank == MASTER_NODE)
+            cout << "\n---------- Start gradient evaluation using sensitivity information ----------" << endl;
+
+        /*--- Definition of the Class for surface deformation ---*/
+
+        surface_movement[iZone] = new CSurfaceMovement();
+
+        /*--- Copy coordinates to the surface structure ---*/
+
+        surface_movement[iZone]->CopyBoundary(geometry_container[iZone][INST_0], config_container[iZone]);
+
+        /*--- If AD mode is enabled we can use it to compute the projection,
+         *    otherwise we use finite differences. ---*/
+
+        if (config_container[iZone]->GetAD_Mode())
+            SetDiscAdjDeformation_AD(geometry_container[iZone][INST_0], config_container[iZone], surface_movement[iZone] , Gradient);
+        else
+            SetDiscAdjDeformation_FD(geometry_container[iZone][INST_0], config_container[iZone], surface_movement[iZone] , Gradient);
+
+        }
+    } // for iZone
+
+    /*--- Write the gradient to a file ---*/
+
+    if (rank == MASTER_NODE)
+        Gradient_file.open(config_container[ZONE_0]->GetObjFunc_Grad_FileName().c_str(), ios::out);
+
+    /*--- Print gradients to screen and writes to file ---*/
+
+    OutputGradient(Gradient, config_container[ZONE_0], Gradient_file);
+
+}
+
+void CDiscAdjDeformationDriver::Postprocessing() {
+
+    for (auto iDV = 0u; iDV  < config_container[ZONE_0]->GetnDV(); iDV++){
+          delete [] Gradient[iDV];
+    }
+    delete [] Gradient;
+
+    delete driver_config;
+    driver_config = nullptr;
+
+    if (rank == MASTER_NODE)
+        cout << "\n------------------------- Solver Postprocessing -------------------------" << endl;
+
+    if (geometry_container != nullptr) {
+        for (iZone = 0; iZone < nZone; iZone++) {
+            if (geometry_container[iZone] != nullptr) {
+                for (iInst = 0; iInst < nInst[iZone]; iInst++){
+                    if (geometry_container[iZone][iInst] != nullptr) {
```

Comment:
These inner-most null-checks are not needed and we have been removing them from the rest of the code.